validator_manager/src/move_validators.rs

```diff
@@ -0,0 +1,1261 @@
+                "all" => Validators::All,
+                pubkeys => pubkeys
+                    .split(',')
+                    .map(PublicKeyBytes::from_str)
+                    .collect::<Result<Vec<_>, _>>()
+                    .map(Validators::Specific)?,
+            },
+            (None, None) => Err(format!(
+                "Must supply either --{VALIDATORS_FLAG} or --{COUNT_FLAG}."
+            ))?,
+            (Some(_), Some(_)) => {
+                Err("Cannot supply both --{VALIDATORS_FLAG} and --{COUNT_FLAG}.")?
+            }
+        };
+
+        Ok(Self {
+            src_vc_url: clap_utils::parse_required(matches, SRC_VC_URL_FLAG)?,
+            src_vc_token_path: clap_utils::parse_required(matches, SRC_VC_TOKEN_FLAG)?,
+            dest_vc_url: clap_utils::parse_required(matches, DEST_VC_URL_FLAG)?,
+            dest_vc_token_path: clap_utils::parse_required(matches, DEST_VC_TOKEN_FLAG)?,
+            validators,
+            builder_proposals: clap_utils::parse_optional(matches, BUILDER_PROPOSALS_FLAG)?,
+            fee_recipient: clap_utils::parse_optional(matches, FEE_RECIPIENT_FLAG)?,
+            gas_limit: clap_utils::parse_optional(matches, GAS_LIMIT_FLAG)?,
+            password_source: PasswordSource::Interactive {
+                stdin_inputs: cfg!(windows) || matches.is_present(STDIN_INPUTS_FLAG),
+            },
+        })
+    }
+}
+
+pub async fn cli_run<'a>(
+    matches: &'a ArgMatches<'a>,
+    dump_config: DumpConfig,
+) -> Result<(), String> {
+    let config = MoveConfig::from_cli(matches)?;
+    if dump_config.should_exit_early(&config)? {
+        Ok(())
+    } else {
+        run(config).await
+    }
+}
+
+async fn run<'a>(config: MoveConfig) -> Result<(), String> {
+    let MoveConfig {
+        src_vc_url,
+        src_vc_token_path,
+        dest_vc_url,
+        dest_vc_token_path,
+        validators,
+        builder_proposals,
+        fee_recipient,
+        gas_limit,
+        mut password_source,
+    } = config;
+
+    // Moving validators between the same VC is unlikely to be useful and probably indicates a user
+    // error.
+    if src_vc_url == dest_vc_url {
+        return Err(format!(
+            "--{} and --{} must be different",
+            SRC_VC_URL_FLAG, DEST_VC_URL_FLAG
+        ));
+    }
+
+    let (src_http_client, src_keystores) =
+        vc_http_client(src_vc_url.clone(), &src_vc_token_path).await?;
+    let (dest_http_client, _dest_keystores) =
+        vc_http_client(dest_vc_url.clone(), &dest_vc_token_path).await?;
+
+    if src_keystores.is_empty() {
+        return Err(NO_VALIDATORS_MSG.to_string());
+    }
+
+    let pubkeys_to_move = match validators {
+        Validators::All => src_keystores.iter().map(|v| v.validating_pubkey).collect(),
+        Validators::Count(count) => {
+            let mut viable_pubkeys: Vec<_> = src_keystores
+                .iter()
+                .filter(|v| !v.readonly.unwrap_or(true))
```

Comment:
There are a couple of places where we assume `readonly: null` means `readonly: true`. I wonder if this could unnecessarily prevent use of the validator-manager with other clients that return `readonly: null`, meaning `readonly: false`. The spec isn't particularly clear about what an absent readonly value means: https://github.com/ethereum/keymanager-APIs/blob/ee334a457a95b5ee0230c2f4603efc45e9830a1f/apis/local_keystores.yaml#L24

I guess we can cross that bridge when we come to it.

Reply:
Oops, intended this as a review comment and was gonna check that we use the standard API. Looks like we're using `delete_lighthouse_keystores`, so no issue with cross-client compat

Reply:
Yeah, I'm not exactly sure what's best here. Assuming `null == true` and logging a message seems like a pretty gentle way to go about it.